(ESP32-S3, ESP32) "Unhandled interrupt on ProCpu"

When testing out the [v0.3.0-rc.1](https://github.com/ariel-os/ariel-os/releases/tag/v0.3.0-rc.1), I've found the following panic again, when running the `http-example` on espressif-esp32-s3-devkitc-1, after the HTML document is successfully fetched:

```
</html> (http_client http-client/src/main.rs:88)
[ERROR] panicked at 'Unhandled interrupt on ProCpu' (esp_hal src/interrupt/mod.rs:90)
```

The panic does *not* happen when removing the `exit()` call at the end of the example, as suggested before by @anlavandier.

--- a/book/src/debug-console.md
+++ b/book/src/debug-console.md
@@ -21,6 +21,14 @@ The laze configuration automatically enables semihosting on the target when the 
 When the flashing tool does not, support for semihosting can still be enabled in the firmware by selecting the `semihosting` [laze module][laze-modules-book].
 This is needed to later be able to attach a semihosting-enabled host tool to the target.
 
+> [!NOTE]
+> When semihosting is enabled on the target and no host tool supporting semihosting (or a debugger) is connected, calling [`ariel_os::debug::exit()`][debug-exit-fn-rustdoc] may result in a panic.
+> For example on ESP using `espflash` you would get:
+>
+> ```
+> [ERROR] panicked at 'Unhandled interrupt on ProCpu' (esp_hal src/interrupt/mod.rs:90)
+> ```
+
 ## Logging
 
 Ariel OS supports logging on all platforms and it is enabled by default with the `logging-facade` [laze module][laze-modules-book].

--- a/laze-project.yml
+++ b/laze-project.yml
@@ -24,7 +24,6 @@ contexts:
       - ?critical-section
       - ?hw/device-identity
       - ?lto
-      - ?semihosting
 
       # Selecting default toolchain here. try xtensa, stable, nightly.
       - ?xtensa
@@ -1160,6 +1159,7 @@ modules:
     help: use probe-rs as runner
     selects:
       - ?debug-console
+      - ?semihosting
       - defmt:
           - defmt-rtt
       - log:
@@ -1202,6 +1202,8 @@ modules:
           - probe-rs reset --chip ${PROBE_RS_CHIP} $@
 
   - name: openocd
+    selects:
+      - ?semihosting
     tasks:
       flash:
         cmd:

